docs/blockentities/index.md

```diff
@@ -1,141 +1,239 @@
+
+Be aware that the `#tick` method is actually called every tick. Due to this, you should avoid doing a lot of complex calculations in here if you can, for example by only calculating things every X ticks, or by caching the results.
+
+## Syncing
 
-// Can override IBlockEntityExtension#onDataPacket. By default, this will defer to  BlockEntity#loadWithComponents.
+Block entity logic is usually run on the server. As such, we need to tell the client what we are doing. There are three ways to do just that: on chunk load, on block update, or by using a custom packet. You should generally only sync information when it is necessary, to not needlessly clog up the network. 
+
+### Syncing on Chunk Load
+
+A chunk is loaded (and by extension, this method is utilized) each time it is read from either network or disk. To send your data here, you need to override the following methods:
+
+```java
+public class MyBlockEntity extends BlockEntity {
+    // ...
+
+    // Create an update tag here. For block entities with only a few fields, this can just call #saveAdditional.
+    @Override
+    public CompoundTag getUpdateTag(HolderLookup.Provider registries) {
+        CompoundTag tag = new CompoundTag();
+        // You can also opt to only save the data that has actually changed here.
+        // This makes sense especially for block entities with larger amounts of data.
+        saveAdditional(tag, registries);
+        return tag;
+    }
+
+    // Handle a received update tag here. The default implementation calls #loadAdditional here,
+    // so you do not need to override this method if you don't plan to do anything beyond that.
+    @Override
+    public void handleUpdateTag(CompoundTag tag, HolderLookup.Provider registries) {
+        super.handleUpdateTag(tag, registries);
+    }
+}
 ```
-The static constructors `ClientboundBlockEntityDataPacket#create` takes:
 
-- The `BlockEntity`.
-- An optional function to get the `CompoundTag` from the `BlockEntity` and a `RegistryAccess`. By default, this uses `BlockEntity#getUpdateTag`.
+### Syncing on Block Update
 
-Now, to send the packet, an update notification must be given on the server.
+This method is used whenever a block update occurs. Block updates must be triggered manually, but are generally processed faster than chunk syncing.
 
 ```java
-Level#sendBlockUpdated(BlockPos pos, BlockState oldState, BlockState newState, int flags)
+public class MyBlockEntity extends BlockEntity {
+    // ...
+
+    // Create an update tag here, like above.
+    @Override
+    public CompoundTag getUpdateTag(HolderLookup.Provider registries) {
+        CompoundTag tag = new CompoundTag();
+        saveAdditional(tag, registries);
+        return tag;
+    }
+
+    // Return our packet here. This method returning a non-null result tells the game to use this packet for syncing.
+    @Override
+    public Packet<ClientGamePacketListener> getUpdatePacket() {
+        return ClientboundBlockEntityDataPacket.create(this);
```

Comment:
This should mention that the single-arg constructor uses `getUpdateTag()` to collect the data to be synced. The second constructor should be mentioned as well with the additional ability to omit unchanged data as mentioned before.